### DISCUSSION TITLE
How to understand response_time in socketio engine

### DISCUSSION THREAD

**Discussion by bkmalan** [artilleryio/artillery#2712] [Artillery]
Hi,
I'm using artillery to test my load test my socketio service. In multiple tests that I've run the response time seems to be around 10000 and I'm not sure how to understand that. Any inputs on this is greatly appreciated.
Kudos to all the moderators for maintaining this great package.

I've attached a screenshot of my response
![image](https://github.com/artilleryio/artillery/assets/55806149/3ff78ee8-ed24-4f06-9fbf-00188f87d558)

**Comment by bernardobridge** [comment]
Hello @bkmalan , can you please share your yaml test script?

**Reply by bkmalan** [reply]
Sure. Here are the config and scenario files. The above screenshot is the report of running it on my dev machine
**Config:**

config:
  processor: "./processor.js"
  environments:
    dev:
      target: "http://localhost:4000"
      phases:
        - duration: 10
          arrivalRate: 1
    qa:
      target: "<qa-url>"
      phases:
        - name: "Warm up"
          duration: 120
          arrivalRate: 3
          rampTo: 20
  plugins:
    publish-metrics:
      - type: newrelic
        licenseKey: "{{ $env.NEWRELIC_API_KEY }}"
        prefix: "artillery-test-{{ $environment }}"
        attributes:
          - "type:soak-test"
          - "service:my-service"
        event:
          accountId: "accId"
          eventType: "Artillery_load_test"
          attributes:
            - "alertType:info"
            - "priority:low"
            - "testId:myTest123"
  socketio:
    transports: ["websocket"]
before:
  flow:
    - function: "prepareLogin"
    - log: "the apiurl is {{ apiUrl }}/login {{ email }} {{ password }}"
    - post:
        url: "{{ apiUrl }}/login"
        afterResponse: "after"
        json:
          email: "{{ email }}"
          password: "{{ password }}"
        capture:
          - json: "$.entity.token"
            as: "token"
    - log: "Got token: {{ token }} {{ email }} {{ password }}"



**Scenario**


scenarios:
  - name: Global Search
    weight: 3
    engine: socketio
    socketio:
      extraHeaders:
        authorization: "{{ token }}"
    flow:
      - function: "generateSearchPhrase"
      - log: "this is token {{ token }}"
      - log: "Searching for {{ phrase }} {{ value }} {{ category }}"
      - emit:
          channel: /global-search
          data:
            phrase: "{{ phrase }}"
            token: "{{ token }}"
        namespace: /events/
        response:
          on: /global-search
          match:
            - json: "$.0.results.0.category"
              value: "{{ category }}"
            - json: "$.0.results.0.items.0.ticker"
              value: "{{ value }}"

**Reply by davidbarkercv** [reply]
Im also encountering this issue. As soon as i put the repsonse object on the time increases to 10 seconds (Which is also the timeout) Response from my socketio API are quick.

socketio.emit: ................................................................. 5
socketio.emit_rate: ............................................................ 1/sec
socketio.response_time:
  min: ......................................................................... 10000
  max: ......................................................................... 10012.7
  mean: ........................................................................ 10002.9
  median: ...................................................................... 9999.2
  p95: ......................................................................... 9999.2
  p99: ......................................................................... 9999.2
vusers.created: ................................................................ 5
vusers.created_by_name.Example: .......................... 5
vusers.failed: ................................................................. 5

So either im misunderstanding what socketio.response_time: is?

scenarios:
  - name: "Example"
    engine: socketio
    flow:
      
      - emit:
          channel: "va"
          data:
            action: "POST"
            endpoint: "session"
            payload:
              "business-area": "XYZ"
              channel: "XYZ"
        response:
          channel: "var"
          capture:
            - json: "$[0].data.session.token"
              as: "authToken"
